packages/office-ui-fabric-react/src/components/TextField/TextField.test.tsx

```diff
@@ -495,6 +495,41 @@ describe('TextField', () => {
     expect(textField.getDOMNode().querySelector('textarea')!.value).toEqual('initial value');
   });
 
+  it('should update value when defaultValue changes and value prop is not set', () => {
+    const defaultValue1 = 'default value 1';
+    const defaultValue2 = 'default value 2';
+
+    const textField = mount(<TextField defaultValue={defaultValue1} />);
+    expect(textField.getDOMNode().querySelector('input')).toBeTruthy();
+    expect(textField.getDOMNode().querySelector('input')!.value).toEqual(defaultValue1);
+
+    textField.setProps({ defaultValue: defaultValue2 });
+    expect(textField.getDOMNode().querySelector('input')).toBeTruthy();
+    expect(textField.getDOMNode().querySelector('input')!.value).toEqual(defaultValue2);
+
+    textField.setProps({ defaultValue: undefined });
+    expect(textField.getDOMNode().querySelector('input')).toBeTruthy();
+    expect(textField.getDOMNode().querySelector('input')!.value).toEqual('');
+  });
+
+  it('should not update value when defaultValue changes and value prop is set', () => {
+    const defaultValue1 = 'default value 1';
+    const defaultValue2 = 'default value 2';
+    const testValue = 'test value';
+
+    const textField = mount(<TextField defaultValue={defaultValue1} />);
+    expect(textField.getDOMNode().querySelector('input')).toBeTruthy();
+    expect(textField.getDOMNode().querySelector('input')!.value).toEqual(defaultValue1);
+
+    textField.setProps({ value: testValue, defaultValue: defaultValue2 });
+    expect(textField.getDOMNode().querySelector('input')).toBeTruthy();
+    expect(textField.getDOMNode().querySelector('input')!.value).toEqual(testValue);
```

Comment:
I know pulling out values that are used multiple times as a dev seems to do the right thing to do. Personally when it comes to test code i find it easier to read when just repeating the values.

```ts
    textField.setProps({ value: 'John', defaultValue: 'Your name' });

    expect(textField.getDOMNode().querySelector('input')).toBeTruthy();
    expect(textField.getDOMNode().querySelector('input')!.value).toEqual('John');
```

Additionally it improves readability a bit more even when you put a new line between the stuff you do and the things you assert.
In the case above the `setProps` and `expect`.